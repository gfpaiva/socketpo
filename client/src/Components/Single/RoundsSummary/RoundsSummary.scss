@import '../../../Scss/root';

.single {
	&__play {
		&-summary-result {
			strong { @include responsive-font-size(32px) }
		}

		&-results {
			display: flex;
			flex-wrap: wrap;
			justify-content: center;
			text-align: center;
		}

		&-result {
			@media screen and (max-width: 767px) { display: none }

			svg {
				height: 40px;
				width: 40px;
			}
		}
	}
}
