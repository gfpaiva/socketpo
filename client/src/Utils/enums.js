import React from 'react';
import Avatar from 'avataaars';

import {
	Rock,
	Paper,
	Scissors,
} from '../Components/Icons/Icons';

const statusValues = {
	0: 'Stand By',
	1: 'In progress',
	2: 'Finished'
};

const playValues = {
	0: 'Miss Play',
	1: 'Rock',
	2: 'Paper',
	3: 'Scissors'
};

const playIcons = {
	1: <Rock />,
	2: <Paper />,
	3: <Scissors />
}

const avatarValues = {
	1: <Avatar
		topType='LongHairMiaWallace'
		accessoriesType='Prescription02'
		hairColor='BrownDark'
		facialHairType='Blank'
		clotheType='Hoodie'
		clotheColor='PastelBlue'
		eyeType='Happy'
		eyebrowType='Default'
		mouthType='Smile'
		skinColor='Light'
	/>,
	2: <Avatar
		topType='LongHairBigHair'
		accessoriesType='Blank'
		hairColor='Blonde'
		facialHairType='Blank'
		clotheType='ShirtVNeck'
		clotheColor='PastelYellow'
		eyeType='Wink'
		eyebrowType='RaisedExcitedNatural'
		mouthType='Serious'
		skinColor='DarkBrown'
	/>,
	3: <Avatar
		topType='ShortHairShortFlat'
		accessoriesType='Prescription01'
		hairColor='Blonde'
		facialHairType='Blank'
		clotheType='BlazerSweater'
		eyeType='Default'
		eyebrowType='DefaultNatural'
		mouthType='Serious'
		skinColor='Pale'
	/>,
	4: <Avatar
		topType='ShortHairShortWaved'
		accessoriesType='Blank'
		hairColor='Auburn'
		facialHairType='BeardLight'
		facialHairColor='Red'
		clotheType='BlazerShirt'
		clotheColor='Gray01'
		eyeType='WinkWacky'
		eyebrowType='UnibrowNatural'
		mouthType='Grimace'
		skinColor='DarkBrown'
	/>,
};

export const parseStatus = statusNum => statusValues[statusNum];
export const parsePlay = playNum => playValues[playNum];
export const parsePlayIcons = playNum => playIcons[playNum];
export const parseAvatar = avatarNum => avatarValues[avatarNum];
