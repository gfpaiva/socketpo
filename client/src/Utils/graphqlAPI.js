import gql from 'graphql-tag';

export const totalGameFields = `
name
hash
status
players {
	id
	name
	avatar
	ready
	roundsWin
}
results {
	message
	matchWinner {
		id
		name
		avatar
	}
	rounds {
		isDraw
		finished
		plays {
			player {
				id
				name
			}
			play
		}
		winner {
			id
			name
			avatar
		}
	}
}`;

export const getGame = gql`
	query getGame($hash: String!) {
		GameByHash(hash: $hash) {
			${totalGameFields}
		}
	}
`;

export const play = gql`
	mutation play($hash: String!, $player: PlayerInput!, $play: Int! ) {
		play(hash: $hash, player: $player, play: $play ) {
			hash
			status
		}
	}
`;

export const gameSub = gql`
	subscription gameSub($hash: String!) {
		gameSubscription(hash: $hash) {
			game {
				${totalGameFields}
			}
			message
			hash
			player
		}
	}
`;
