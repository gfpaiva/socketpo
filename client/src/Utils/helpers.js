export const sanitize = str => (str.toLowerCase()
.replace(/[àáâãäå]/ig, 'a')
.replace(/[éèëê]/ig, 'e')
.replace(/[íìïî]/ig, 'i')
.replace(/[óòõöô]/ig, 'o')
.replace(/[uúùüû]/ig, 'u')
.replace(/[ç]/ig, 'c')
.replace(/^[0-9]/ig, '')
.replace(/\s/ig, '-'));
