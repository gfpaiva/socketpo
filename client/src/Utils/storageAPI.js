export const setObject = (key, object) => localStorage.setItem(key, JSON.stringify(object));
export const getObject = key => JSON.parse(localStorage.getItem(key));
