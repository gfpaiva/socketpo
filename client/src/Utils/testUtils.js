export const fakePlayer = (rewrite) => ({
	__typename: 'Player',
	avatar: 3,
	id: 'player01',
	name: 'Player 01',
	ready: false,
	roundsWin: 0,
	...rewrite
});

export const fakeGame = (fullPlayers = true, matchWinner = null, rounds = [], rewrite) => {

	const players = fullPlayers ? [fakePlayer(), fakePlayer({ id: 'player02', name: 'Player 02' })] : [fakePlayer()]

	return {
		__typename: 'Game',
		id: 'game01',
		hash: 'game01',
		name: 'Game 01',
		players,
		results: {
			__typename: "Result",
			matchWinner,
			message: null,
			rounds
		},
		status: 0,
		...rewrite
	};
};

export const fakeRound = (rewrite) => ({
	__typename: 'Round',
	isDraw: false,
	finished: true,
	plays: [
		fakePlay({ play: 1 }),
		fakePlay({ player: fakePlayer({ id: 'player02', name: 'Player 02' }), play: 3 })
	],
	winner: fakePlayer(),
	...rewrite
});

export const fakePlay = (rewrite) => ({
	__typename: 'Play',
	player: fakePlayer(),
	play: 0,
	...rewrite
});
