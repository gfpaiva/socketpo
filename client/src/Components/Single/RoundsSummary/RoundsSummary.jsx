import React from 'react';

import { graphql, compose } from 'react-apollo';
import { withRouter } from 'react-router-dom';

import { getGame } from '../../../Utils/graphqlAPI';

import {
	Draw,
	Win,
	Loose
} from '../../Icons/Icons';

import './RoundsSummary.scss';

const RoundsSummary = ( { getGame, player } ) => {

	const game = getGame.GameByHash;
	const rounds = game && game.results.rounds;

	return (
		<div>
			<p className="single__play-summary-result">
				<strong>{player.roundsWin}</strong>
			</p>

			{rounds.length > 0 && (
				<div className='single__play-results'>
					{rounds.map((round, roundIdx) => {
						if(round.finished) {
							return (
								<div className='card single__play-result' key={`round-${roundIdx}`}>
									<h3>Round {roundIdx+1}</h3>
									<span>
										{round.isDraw && <Draw />}
										{!round.isDraw && round.winner.id === player.id && <Win />}
										{!round.isDraw && round.winner.id !== player.id && <Loose />}
									</span>
								</div>
							);
						}

						return null;
					})}
				</div>
			)}
		</div>
	);
};

export default withRouter(compose(
	graphql(getGame, {
		name: 'getGame',
		options: props => ({ variables: { hash: props.match.params.hash } })
	}),
)(RoundsSummary));
