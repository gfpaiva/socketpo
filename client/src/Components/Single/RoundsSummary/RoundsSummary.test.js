import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import RoundsSummary from './RoundsSummary';
import { getGame } from '../../../Utils/graphqlAPI';
import { fakeGame, fakePlayer, fakeRound } from '../../../Utils/testUtils';

const game = fakeGame(
	true,
	null,
	[
		fakeRound(),
		fakeRound({ winner: fakePlayer({ id: 'player02', name: 'Player 02' }) }),
		fakeRound({ isDraw: true })
	]
);
const player = fakePlayer();
const mocks = [
	{
		request: { query: getGame, variables: { hash: game.hash } },
		result: {
			data: {
				GameByHash: {
					...game
				}
			}
		},
	},
];

describe('<RoundsSummary />', () => {

	it('should mount properly (Win, Lose, Draw)', async () => {
		const wrapper = mount(
			<MockedProvider mocks={mocks}>
				<RoundsSummary.WrappedComponent
					match={{params: {hash: game.hash}}}
					player={player}

				/>
			</MockedProvider>
		);

		await wait();
		wrapper.update();

		expect(wrapper.find('[data-test="summary"]')).toMatchSnapshot();
	});
});
