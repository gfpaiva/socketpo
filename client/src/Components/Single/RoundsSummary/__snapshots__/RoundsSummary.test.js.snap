// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundsSummary /> should mount properly (Win, Lose, Draw) 1`] = `
<div
  data-test="summary"
>
  <p
    className="single__play-summary-result"
  >
    <strong>
      0
    </strong>
  </p>
  <div
    className="single__play-results"
  >
    <div
      className="card single__play-result"
      key="round-0"
    >
      <h3>
        Round 
        1
      </h3>
      <span>
        <Win>
          <svg
            version="1.1"
            viewBox="0 0 91 113.75"
            x="0px"
            y="0px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <path
                d="M23.3022,0.5051 L23.3022,8.0051 L30.8022,8.0051 L38.3022,8.0051 L45.8022,8.0051 L53.3022,8.0051 L60.8022,8.0051 L68.3022,8.0051 L68.3022,0.5051 L60.8022,0.5051 L53.3022,0.5051 L45.8022,0.5051 L38.3022,0.5051 L30.8022,0.5051 L23.3022,0.5051 L23.3022,0.5051 Z M68.3022,8.0051 L68.3022,15.5051 L75.8022,15.5051 L75.8022,8.0051 L68.3022,8.0051 L68.3022,8.0051 Z M75.8022,15.5051 L75.8022,23.0051 L83.3022,23.0051 L83.3022,15.5051 L75.8022,15.5051 L75.8022,15.5051 Z M83.3022,23.0051 L83.3022,30.5051 L83.3022,38.0051 L83.3022,45.5051 L83.3022,53.0051 L83.3022,60.5051 L83.3022,68.0051 L90.8022,68.0051 L90.8022,60.5051 L90.8022,53.0051 L90.8022,45.5051 L90.8022,38.0051 L90.8022,30.5051 L90.8022,23.0051 L83.3022,23.0051 L83.3022,23.0051 Z M83.3022,68.0051 L75.8022,68.0051 L75.8022,75.5051 L83.3022,75.5051 L83.3022,68.0051 L83.3022,68.0051 Z M75.8022,75.5051 L68.3022,75.5051 L68.3022,83.0051 L75.8022,83.0051 L75.8022,75.5051 L75.8022,75.5051 Z M68.3022,83.0051 L60.8022,83.0051 L53.3022,83.0051 L45.8022,83.0051 L38.3022,83.0051 L30.8022,83.0051 L23.3022,83.0051 L23.3022,90.5051 L30.8022,90.5051 L38.3022,90.5051 L45.8022,90.5051 L53.3022,90.5051 L60.8022,90.5051 L68.3022,90.5051 L68.3022,83.0051 L68.3022,83.0051 Z M23.3022,83.0051 L23.3022,75.5051 L15.8022,75.5051 L15.8022,83.0051 L23.3022,83.0051 L23.3022,83.0051 Z M15.8022,75.5051 L15.8022,68.0051 L8.3022,68.0051 L8.3022,75.5051 L15.8022,75.5051 L15.8022,75.5051 Z M8.3022,68.0051 L8.3022,60.5051 L8.3022,53.0051 L8.3022,45.5051 L8.3022,38.0051 L8.3022,30.5051 L8.3022,23.0051 L0.8022,23.0051 L0.8022,30.5051 L0.8022,38.0051 L0.8022,45.5051 L0.8022,53.0051 L0.8022,60.5051 L0.8022,68.0051 L8.3022,68.0051 L8.3022,68.0051 Z M8.3022,23.0051 L15.8022,23.0051 L15.8022,15.5051 L8.3022,15.5051 L8.3022,23.0051 L8.3022,23.0051 Z M15.8022,15.5051 L23.3022,15.5051 L23.3022,8.0051 L15.8022,8.0051 L15.8022,15.5051 L15.8022,15.5051 Z M75.8022,23.0051 L68.3022,23.0051 L68.3022,30.5051 L60.8022,30.5051 L60.8022,38.0051 L53.3022,38.0051 L53.3022,45.5051 L45.8022,45.5051 L45.8022,53.0051 L38.3022,53.0051 L38.3022,45.5051 L30.8022,45.5051 L30.8022,38.0051 L23.3022,38.0051 L23.3022,45.5051 L23.3022,53.0051 L30.8022,53.0051 L30.8022,60.5051 L38.3022,60.5051 L38.3022,68.0051 L45.8022,68.0051 L45.8022,60.5051 L53.3022,60.5051 L53.3022,53.0051 L60.8022,53.0051 L60.8022,45.5051 L68.3022,45.5051 L68.3022,38.0051 L75.8022,38.0051 L75.8022,30.5051 L75.8022,23.0051 L75.8022,23.0051 Z"
                fill="#000000"
              />
            </g>
          </svg>
        </Win>
      </span>
    </div>
    <div
      className="card single__play-result"
      key="round-1"
    >
      <h3>
        Round 
        2
      </h3>
      <span>
        <Lose>
          <svg
            version="1.1"
            viewBox="0 0 91 113.75"
            x="0px"
            y="0px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <path
                d="M23.478,0.45508 L23.478,7.95508 L30.978,7.95508 L38.478,7.95508 L45.978,7.95508 L53.478,7.95508 L60.978,7.95508 L68.478,7.95508 L68.478,0.45508 L60.978,0.45508 L53.478,0.45508 L45.978,0.45508 L38.478,0.45508 L30.978,0.45508 L23.478,0.45508 L23.478,0.45508 Z M68.478,7.95508 L68.478,15.45508 L75.978,15.45508 L75.978,7.95508 L68.478,7.95508 L68.478,7.95508 Z M75.978,15.45508 L75.978,22.95508 L83.478,22.95508 L83.478,15.45508 L75.978,15.45508 L75.978,15.45508 Z M83.478,22.95508 L83.478,30.45508 L83.478,37.95508 L83.478,45.45508 L83.478,52.95508 L83.478,60.45508 L83.478,67.95508 L90.978,67.95508 L90.978,60.45508 L90.978,52.95508 L90.978,45.45508 L90.978,37.95508 L90.978,30.45508 L90.978,22.95508 L83.478,22.95508 L83.478,22.95508 Z M83.478,67.95508 L75.978,67.95508 L75.978,75.45508 L83.478,75.45508 L83.478,67.95508 L83.478,67.95508 Z M75.978,75.45508 L68.478,75.45508 L68.478,82.95508 L75.978,82.95508 L75.978,75.45508 L75.978,75.45508 Z M68.478,82.95508 L60.978,82.95508 L53.478,82.95508 L45.978,82.95508 L38.478,82.95508 L30.978,82.95508 L23.478,82.95508 L23.478,90.45508 L30.978,90.45508 L38.478,90.45508 L45.978,90.45508 L53.478,90.45508 L60.978,90.45508 L68.478,90.45508 L68.478,82.95508 L68.478,82.95508 Z M23.478,82.95508 L23.478,75.45508 L15.978,75.45508 L15.978,82.95508 L23.478,82.95508 L23.478,82.95508 Z M15.978,75.45508 L15.978,67.95508 L8.478,67.95508 L8.478,75.45508 L15.978,75.45508 L15.978,75.45508 Z M8.478,67.95508 L8.478,60.45508 L8.478,52.95508 L8.478,45.45508 L8.478,37.95508 L8.478,30.45508 L8.478,22.95508 L0.978,22.95508 L0.978,30.45508 L0.978,37.95508 L0.978,45.45508 L0.978,52.95508 L0.978,60.45508 L0.978,67.95508 L8.478,67.95508 L8.478,67.95508 Z M8.478,22.95508 L15.978,22.95508 L15.978,15.45508 L8.478,15.45508 L8.478,22.95508 L8.478,22.95508 Z M15.978,15.45508 L23.478,15.45508 L23.478,7.95508 L15.978,7.95508 L15.978,15.45508 L15.978,15.45508 Z M23.478,22.95508 L23.478,30.45508 L30.978,30.45508 L30.978,22.95508 L23.478,22.95508 L23.478,22.95508 Z M30.978,30.45508 L30.978,37.95508 L38.478,37.95508 L38.478,30.45508 L30.978,30.45508 L30.978,30.45508 Z M38.478,37.95508 L38.478,45.45508 L38.478,52.95508 L45.978,52.95508 L53.478,52.95508 L53.478,45.45508 L53.478,37.95508 L45.978,37.95508 L38.478,37.95508 L38.478,37.95508 Z M53.478,37.95508 L60.978,37.95508 L60.978,30.45508 L53.478,30.45508 L53.478,37.95508 L53.478,37.95508 Z M60.978,30.45508 L68.478,30.45508 L68.478,22.95508 L60.978,22.95508 L60.978,30.45508 L60.978,30.45508 Z M53.478,52.95508 L53.478,60.45508 L60.978,60.45508 L60.978,52.95508 L53.478,52.95508 L53.478,52.95508 Z M60.978,60.45508 L60.978,67.95508 L68.478,67.95508 L68.478,60.45508 L60.978,60.45508 L60.978,60.45508 Z M38.478,52.95508 L30.978,52.95508 L30.978,60.45508 L38.478,60.45508 L38.478,52.95508 L38.478,52.95508 Z M30.978,60.45508 L23.478,60.45508 L23.478,67.95508 L30.978,67.95508 L30.978,60.45508 L30.978,60.45508 Z"
                fill="#000000"
              />
            </g>
          </svg>
        </Lose>
      </span>
    </div>
    <div
      className="card single__play-result"
      key="round-2"
    >
      <h3>
        Round 
        3
      </h3>
      <span>
        <Draw>
          <svg
            version="1.1"
            viewBox="0 0 91 113.75"
            x="0px"
            y="0px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <path
                d="M22.8408,0.5051 L22.8408,8.0051 L30.3408,8.0051 L37.8408,8.0051 L45.3408,8.0051 L52.8408,8.0051 L60.3408,8.0051 L67.8408,8.0051 L67.8408,0.5051 L60.3408,0.5051 L52.8408,0.5051 L45.3408,0.5051 L37.8408,0.5051 L30.3408,0.5051 L22.8408,0.5051 L22.8408,0.5051 Z M67.8408,8.0051 L67.8408,15.5051 L75.3408,15.5051 L75.3408,8.0051 L67.8408,8.0051 L67.8408,8.0051 Z M75.3408,15.5051 L75.3408,23.0051 L82.8408,23.0051 L82.8408,15.5051 L75.3408,15.5051 L75.3408,15.5051 Z M82.8408,23.0051 L82.8408,30.5051 L82.8408,38.0051 L82.8408,45.5051 L82.8408,53.0051 L82.8408,60.5051 L82.8408,68.0051 L90.3408,68.0051 L90.3408,60.5051 L90.3408,53.0051 L90.3408,45.5051 L90.3408,38.0051 L90.3408,30.5051 L90.3408,23.0051 L82.8408,23.0051 L82.8408,23.0051 Z M82.8408,68.0051 L75.3408,68.0051 L75.3408,75.5051 L82.8408,75.5051 L82.8408,68.0051 L82.8408,68.0051 Z M75.3408,75.5051 L67.8408,75.5051 L67.8408,83.0051 L75.3408,83.0051 L75.3408,75.5051 L75.3408,75.5051 Z M67.8408,83.0051 L60.3408,83.0051 L52.8408,83.0051 L45.3408,83.0051 L37.8408,83.0051 L30.3408,83.0051 L22.8408,83.0051 L22.8408,90.5051 L30.3408,90.5051 L37.8408,90.5051 L45.3408,90.5051 L52.8408,90.5051 L60.3408,90.5051 L67.8408,90.5051 L67.8408,83.0051 L67.8408,83.0051 Z M22.8408,83.0051 L22.8408,75.5051 L15.3408,75.5051 L15.3408,83.0051 L22.8408,83.0051 L22.8408,83.0051 Z M15.3408,75.5051 L15.3408,68.0051 L7.8408,68.0051 L7.8408,75.5051 L15.3408,75.5051 L15.3408,75.5051 Z M7.8408,68.0051 L7.8408,60.5051 L7.8408,53.0051 L7.8408,45.5051 L7.8408,38.0051 L7.8408,30.5051 L7.8408,23.0051 L0.3408,23.0051 L0.3408,30.5051 L0.3408,38.0051 L0.3408,45.5051 L0.3408,53.0051 L0.3408,60.5051 L0.3408,68.0051 L7.8408,68.0051 L7.8408,68.0051 Z M7.8408,23.0051 L15.3408,23.0051 L15.3408,15.5051 L7.8408,15.5051 L7.8408,23.0051 L7.8408,23.0051 Z M15.3408,15.5051 L22.8408,15.5051 L22.8408,8.0051 L15.3408,8.0051 L15.3408,15.5051 L15.3408,15.5051 Z M30.3408,23.0051 L30.3408,30.5051 L37.8408,30.5051 L37.8408,23.0051 L30.3408,23.0051 L30.3408,23.0051 Z M30.3408,30.5051 L22.8408,30.5051 L22.8408,38.0051 L30.3408,38.0051 L30.3408,30.5051 L30.3408,30.5051 Z M22.8408,38.0051 L15.3408,38.0051 L15.3408,45.5051 L15.3408,53.0051 L22.8408,53.0051 L22.8408,45.5051 L22.8408,38.0051 L22.8408,38.0051 Z M22.8408,53.0051 L22.8408,60.5051 L30.3408,60.5051 L30.3408,53.0051 L22.8408,53.0051 L22.8408,53.0051 Z M30.3408,60.5051 L30.3408,68.0051 L37.8408,68.0051 L37.8408,60.5051 L30.3408,60.5051 L30.3408,60.5051 Z M52.8408,23.0051 L52.8408,30.5051 L60.3408,30.5051 L60.3408,23.0051 L52.8408,23.0051 L52.8408,23.0051 Z M60.3408,30.5051 L60.3408,38.0051 L67.8408,38.0051 L67.8408,30.5051 L60.3408,30.5051 L60.3408,30.5051 Z M67.8408,38.0051 L67.8408,45.5051 L67.8408,53.0051 L75.3408,53.0051 L75.3408,45.5051 L75.3408,38.0051 L67.8408,38.0051 L67.8408,38.0051 Z M67.8408,53.0051 L60.3408,53.0051 L60.3408,60.5051 L67.8408,60.5051 L67.8408,53.0051 L67.8408,53.0051 Z M60.3408,60.5051 L52.8408,60.5051 L52.8408,68.0051 L60.3408,68.0051 L60.3408,60.5051 L60.3408,60.5051 Z"
                fill="#000000"
              />
            </g>
          </svg>
        </Draw>
      </span>
    </div>
  </div>
</div>
`;
